here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/storage/ContextWriteFeatures.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.core.models.storage;
+
+import com.here.naksha.lib.core.NakshaVersion;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Abstract class representing WriteFeatures request alongwith list of features as context and list of violations.
+ * Implementing class will define actual data type of context and violations.
+ */
+@ApiStatus.AvailableSince(NakshaVersion.v2_0_11)
+public abstract class ContextWriteFeatures<
+        FEATURE,
+        CTX_TYPE,
```

Comment:
Why do we need context? Wouldn't features and violations be enough for our needs?

Reply:
context is required as per design. Not for mock though.